EIPS/eip-7685.md

```diff
@@ -13,64 +13,61 @@ created: 2024-04-14
 ## Abstract
 
 This proposal defines a general purpose framework for storing contract-triggered
-requests. It extends the execution header and body with a single field each to
-store the request information. This inherently exposes the requests to the
-consensus layer, which can then process each one.
+requests. It extends the execution header with a single field as a commitment to
+the request. Requests are later on exposed to the consensus layer via the
+Engine API.
+
+Requests are not stored on the execution block, only a commitment to the requests
+is stored in the execution block.
 
 ## Motivation
 
 The proliferation of smart contract controlled validators has caused there to be
 a demand for additional EL triggered behaviors. By allowing these systems to
 delegate administrative operations to their governing smart contracts, they can
 avoid intermediaries needing to step in and ensure certain operations occur.
-This creates a safer system for end users.
+This creates a safer system for end users. By abstracting each individual request
+details from the EL, adding new request types is simpler and does not require an
+update on the execution block structure.
 
 ## Specification
 
 ### Execution Layer
 
-#### Request
+#### Requests
+
+Let `request_data` be the output of a system call to a Request Smart Contract.
 
-A `request` consists of a `request_type` prepended to an opaque byte array
-`request_data`:
+`request_data` consists of two parts: `request_type` (single byte) and
+`request_list_ssz` (ssz list encoding of many requests) as following:
 
 ```
-request = request_type ++ request_data
+request_data = request_type ++ request_list_ssz
 ```
 
-Let `requests` be the list of all `request` objects in the block in ascending
-order by type. For example:
+Let `requests` be the concatenation of each `request_data` returned from each request
+contract:
 
 ```
-[0x00_request_0, 0x01_request_0, 0x01_request_1, 0x02_request_0, ...]
+requests = request_data_0 ++ request_data_1 ++ ... ++ request_data_n
 ```
 
-The ordering of requests within a type is to be defined by each request type.
-
-#### Block structure
+Note: each `request_data` represents a different list of requests from a particular
+smart contract, prepended by their type. They must be ordered by their first byte
+(the `request_type`).
 
-The block body is appended with a list of requests. RLP encoding of the extended
-block body structure is computed as follows:
-
-```python
-block_body_rlp = rlp([
-    field_0,
-    ...,
-    # Latest block body field before `requests`
-    field_n,
-    [request_0, ..., request_k],
-])
-```
+The ordering of requests within a type is to be defined by each request type and it
+is the smart contract responsibility to choose.
 
 #### Block Header
 
 Extend the header with a new 32 byte value `requests_hash`:
```

Comment:
IMO, it is important to add that matching the requests obtained from the block execution to the `requests_hash` from the block header is a new block validity condition